packages/zoe/src/zoe.js

```diff
@@ -290,13 +287,19 @@ import { makeTables } from './state';
  * @param {OfferHandle} offerHandle
  * @returns {OfferOutcome}
  *
- *
  * @callback AddNewIssuer
  * Informs Zoe about an issuer and returns a promise for acknowledging
  * when the issuer is added and ready.
  * @param {Promise<Issuer>|Issuer} issuerP Promise for issuer
  * @param {Keyword} keyword Keyword for added issuer
  * @returns {Promise<IssuerRecord>} Issuer is added and ready
+ *
+ * @callback UpdatePublicAPI
+ * Update the publicAPI for the contract instance, as stored by Zoe in
+ * the instanceRecord.
```

Comment:
The description should say "set or update". I think the name of the API function should actually be `setPublicAPI`. Its most common use will be setting it once and only once. 

It could be used to update the API, though we don't have any examples of that yet, do we?

Reply:
That's a good point. I think I had the table functionality in mind. It has an `update` method which is what is being used under the hood to change any values in the instanceRecord, such as the publicAPI. Additionally, we have internal methods like `updateAmounts` (probably needs to be renamed to `updateAllocation`, but that's a side note). The publicAPI does already exist in the instanceRecord, it's set to `undefined`, though. 

So overall, I think `update` isn't out of place, especially since the method could be used again. Is `set` better from a user perspective, do you think?

Reply:
The more I think about it, the less I like having this be generally mutable. Better I think for it to make a one time monotonic transition from being absent to present. You should never need to worry that you might send messages to a stale public API rather than the current one.

As a set once, we should call it neither `set` nor `update` but `init`.